File: monolitic-services/BILL_PAYMENT_SYSTEM/src/main/java/com/cognizant/bill_payment_system/service/BillService.java
package com.cognizant.bill_payment_system.service;

import java.util.List;


import com.cognizant.bill_payment_system.model.Bill;
import com.cognizant.bill_payment_system.model.Payment;

public interface BillService {
	
	Bill getPaymentDetails(String vendorName,String username,String billPaymentGateway);
	List<Bill> getPaymentDetailsOfUser(String vendorName,String username);
	void doPayment(Payment payment);
	
	void addBill(Payment payment);
	
	List<Bill> getUnpaidBillsOfUserAndVendor(String vendorName, String username);
	
	List<Bill> getPaidBillsOfUserAndVendor(String vendorname, String username);
	
	List<Bill> getPaidBillsOfUser(String username);
}

File: webapp/bill-payment-system/src/app/services/bill.service.ts
import { Subject } from 'rxjs';
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { AuthenticateService } from './authentication.service';
import { Bill } from '../model/bill.model';

@Injectable({
    providedIn:'root'
})
export class BillService{
    
    
    token:string='';
    constructor(private httpClient:HttpClient,private authService:AuthenticateService) { }
    
    addBill(bill:Bill){
        this.token=this.authService.accessToken;
        let headers=new HttpHeaders();
        headers=headers.set('Authorization','Bearer '+this.token);
        console.log(bill);
        
        return this.httpClient.post<any>(`http://localhost:8090/payment/bill`,bill,{headers});
      }

      getPendingBillsDetailsByVendorAndUser(vendorName:string,username:string){
        this.token=this.authService.accessToken;
        let headers=new HttpHeaders();
        headers=headers.set('Authorization','Bearer '+this.token);
        return this.httpClient.get(`http://localhost:8090/payment/bill/${vendorName}/${username}`,{headers})
      }
}